bpf/lib/encrypt.h

```diff
@@ -264,15 +267,13 @@ encrypt_overlay_and_redirect(struct __ctx_buff *ctx)
 	if (eth_store_daddr(ctx, &dst_mac, 0) != 0)
 		return DROP_WRITE_ERROR;
 
-	/* need to revalidate data since we just re-wrote mac addresses */
```

Comment:
A little confused about this removal of the validation. 

Once `eth_store_daddr` modifies any offset from the data pointer, won't the verifier invalidate the data pointers? 

I was under the impression if we rewrite any portion of the header, using direct packet access, we must re-validate the packet bounds. Looks like my knowledge if a little off here.

Reply:
Meh, Okay i think I see it. 

You eliminate the need to do another ipv4 header parse and just rewrite the data,data_end pointers with new values.

`vxlan_rewrite_vni` does an internal bounds check. :+1: